plugins/UsersManager/Model.php

```diff
@@ -245,11 +245,14 @@ public function userEmailExists($userEmail)
     public function addUserAccess($userLogin, $access, $idSites)
     {
         foreach ($idSites as $idsite) {
-            $this->getDb()->insert(Common::prefixTable("access"),
-                array("idsite" => $idsite,
-                      "login"  => $userLogin,
-                      "access" => $access)
-            );
+            $count = $this->getDb()->fetchOne("SELECT count(*) FROM " . Common::prefixTable("site"). " WHERE idsite = ?",$idsite);
+            if($count != 0){
+                 $this->getDb()->insert(Common::prefixTable("access"),
+                    array("idsite" => $idsite,
+                         "login"  => $userLogin,
+                         "access" => $access)
+                );
+            }
```

Comment:
Hey @haseebeqx are you still working on the PR? Can I maybe help you somehow or explain it in a different way?